src/utils/internal/isNodeProcess.ts

```diff
@@ -4,7 +4,7 @@
  */
 export function isNodeProcess() {
   // Check browser environment.
-  if (typeof global !== 'object') {
+  if (typeof global !== 'object' || typeof window !== 'undefined') {
```

Comment:
This is not correct: `window` may exist in non-browser environments, such as JSDOM. Such environments must _not_ be treated as a browser, because they cannot have the Service Worker API.

The intention behind `isNodeProcess` function is to return `true` for _any_ environment that is not a native browser. This includes:

- Environments that polyfill browser-like behaviors (JSOM);
- Server-side environments of otherwise browser-oriented tools (runners in Storybook and Cypress);
- Actual Node.js environment (i.e. when developing Node.js servers).

If you think this function doesn't determine a native browser environment properly, such as Electron, could you share more info about when this happens? A reproduction repository would be the best. We may adjust `isNodeProcess` accordingly, but that adjustment won't rely on the existence of `window`.